packages/web3-core/src/web3_config.ts

```diff
@@ -335,7 +336,7 @@ export abstract class Web3Config
 
 	public set maxListenersWarningThreshold(val) {
 		this._triggerConfigChange('maxListenersWarningThreshold', val);
-
+		if (typeof val === 'number') this.setMaxListenerWarningThreshold(val);
```

Comment:
Why only if a number? If it is not a number, try to convert to a number. Max listeners should be a number greater than 0 (Do not set config if it is not)

Reply:
thanks for this, changed